x-pack/test/performance/services/performance.ts

```diff
@@ -166,6 +168,7 @@ export class PerformanceTestingService extends FtrService {
 
   private async tearDown(page: Page, client: CDPSession, context: BrowserContext) {
     if (page) {
+      await sleep(5000);
```

Comment:
1. Could you add a comment *why* it's necessary? With a link to [flushInterval](https://www.elastic.co/guide/en/apm/agent/rum-js/current/configuration.html#flush-interval) config option.
2. Let's maybe configure `flushInterval` explicitly to avoid relying on the default value? Having it configured, we can set the delay slightly above the `flushInterval` value.

Reply:
We are not using rum agent here and we don't have flushInterval on node agent. Ref: https://www.elastic.co/guide/en/apm/agent/nodejs/current/agent-api.html#apm-flush. But I will add relevant comment to explain why do we use flush

Reply:
we don’t use the RUM agent in *this file*, but we can configure `flushInterval` for the RUM agent on Kibana client-side https://github.com/elastic/kibana/blob/34d5ef8fe37f140d444a95f5114cf51ad13e13ae/x-pack/test/performance/config.playwright.ts#L43-L65
and then use the same / a bit longer delay in the `tearDown` step.
Did I miss anything?

Reply:
Anyway, I liked Vighesh' proposal from better
https://github.com/elastic/kibana/pull/129991/files/024d9aad25fe7a7c9cb1f63605e3d4d69ef6321f#r853962293

Reply:
1. That's still the test runners config and I don't know if we are able to provide env variable to change flush interval.
2. I also liked that idea however but I'm not sure about this scenario: let's say there are multiple apm events are being sent it's going to close the page when it gets the first response which is not we want ? @mshustov @vigneshshanmugam maybe it doesn't work like that ?

Reply:
> I also liked that idea however but I'm not sure about this scenario: let's say there are multiple apm events are being sent it's going to close the page when it gets the first response which is not we want ?

It might happen in theory. We can extend this logic to count the number of outcoming and responded requests.
```js
this.rumRequestsInFlight = []

await page.route('**', async (route, request) => {
   if(isRumReportingRequest(route)) {
   rumRequestsInFlight.push(request.response);
});

function tearDown() {
  // wait for the one more reporting
  await page.waitForResponse((response) => response.url().includes('/rum/events'));
  // make sure all the previous requests were finished
  await Promise.all(rumRequestsInFlight);
```
 @vigneshshanmugam is there API to disable metrics collection by the RUM agent and enforce it to report already collected events? 

> maybe it doesn't work like that ?

@suchcodemuchwow how would you suggest this problem be solved?

Reply:
>  let's say there are multiple apm events are being sent it's going to close the page when it gets the first response which is not we want ?
Agreed, if there are multiple transactions in the page within the flush interval, we would need extra guard to count for the no of possible APM server calls from the agent.

The better way to fix this would be 
1. **(Recommended)** - Update the RUM agent to the latest version which makes sure the `events queue` in the agent is flushed before the users leaves the page and we dont need any workarounds here. https://github.com/elastic/apm-agent-rum-js/commit/242981446a6a03ffe3b9bb8c1de34a4b1f9195e9
2. For the time being, I think counting strategy based on the api server calls could work. 
3. (Hacky workaround) - Have not tested it though.
```ts
page.evaluate(() => {
  const apmServer = window.elasticApm.serviceFactory.getService('ApmServer')
  apmServer.queue.flush()
});
```

Reply:
> (Recommended) - Update the RUM agent to the latest version

Opened https://github.com/elastic/kibana/pull/130765